mat64/dense.go

```diff
@@ -136,7 +136,22 @@ func (m *Dense) isolatedWorkspace(a Matrix) (w *Dense, restore func()) {
 	}
 }
 
-func (m *Dense) isZero() bool {
+// Reset zeros the dimensions of the matrix so that it can be reused as the
+// receiver of a dimensionally restricted operation.
+//
+// See the Reseter interface for more information.
+func (m *Dense) Reset() {
+	// Row, Cols and Stride must be zeroed in unison.
+	m.mat.Rows, m.mat.Cols, m.mat.Stride = 0, 0, 0
+	m.capRows, m.capCols = 0, 0
+	m.mat.Data = m.mat.Data[:0]
+}
+
+// IsZero returns whether the receiver is a zero value. A zero value may be
```

Comment:
Maybe: `IsZero returns whether the receiver is zero-sized. Zero-sized matrices can be the receiver for size-restricted operations. Dense matrices can be zeroed using Reset()`